addon/components/infinity-loader.js

```diff
@@ -24,83 +25,32 @@ const InfinityLoaderComponent = Ember.Component.extend({
 
   didInsertElement() {
     this._super(...arguments);
-    this._setupScrollable();
     this.set('guid', Ember.guidFor(this));
-    this._bindEvent('scroll');
-    this._bindEvent('resize');
 
-    if (this.get('_isInfinityPromiseArray')) {
-      this.get('infinityModel').then(() => this._loadMoreIfNeeded());
-    } else {
-      this._loadMoreIfNeeded();
-    }
+    this.setProperties({
+      viewportSpy: true,
+      viewportTolerance: {
+        top: 0,
+        right: 0,
+        bottom: this.get('triggerOffset'),
+        left: 0
+      }
+    });
   },
 
   willDestroyElement() {
     this._super(...arguments);
-    this._unbindEvent('scroll');
-    this._unbindEvent('resize');
+    this._cancelTimers();
   },
 
   _isInfinityPromiseArray: Ember.computed('infinityModel', function() {
     return (this.get('infinityModel') instanceof InfinityPromiseArray);
   }),
 
-  _bindEvent(eventName) {
-    this.get('_scrollable').on(`${eventName}.${this.get('guid')}`, () => {
-      Ember.run.debounce(this, this._loadMoreIfNeeded, this.get('eventDebounce'));
-    });
-  },
-
-  _unbindEvent(eventName) {
-    let scrollable = this.get('_scrollable');
-    if (scrollable) {
-      scrollable.off(`${eventName}.${this.get('guid')}`);
-    }
-  },
-
-  /**
-   * determine how far the infinity-loader component is from the top of the
-    scrollable element
-   * @method _selfOffset
-   * @return {Number}
-   */
-  _selfOffset() {
-    if (this.get('_customScrollableIsDefined')) {
-      return this.$().offset().top - this.get("_scrollable").offset().top + this.get("_scrollable").scrollTop();
-    } else {
-      return this.$().offset().top;
-    }
-  },
-
   /**
-   * total pixels where _scrollable's bottom position is from top of screen
-   * @method _bottomOfScrollableOffset
-   * @return {Number}
+   * @method didEnterViewport
    */
-  _bottomOfScrollableOffset() {
-    return this.get('_scrollable').height() + this.get("_scrollable").scrollTop();
-  },
-
-  /**
-   * distance infinity loader is from top of _scrollable/window's top position
-    less the buffer amount specified by triggerOffset
-   * @method _triggerOffset
-   * @return {Number}
-   */
-  _triggerOffset() {
-    return this._selfOffset() - this.get('triggerOffset');
-  },
-
-  /**
-   * is _scrollable's bottom position greater than the distance from the
-    infinity-loader component to _scrollable's top
-   * in other words if at top of page and _scrollable's height is 100px and
-    infinity loader is 200px from the top, has the scrollable traversed 101px to trigger loading more records
-   * @method _shouldLoadMore
-   * @return {Boolean}
-   */
-  _shouldLoadMore() {
+  didEnterViewport() {
     const consideredReady = !this.get('_isInfinityPromiseArray') || this.get('infinityModel.isFulfilled');
     if (
```

Comment:
I'm thinking we still want this check here for the non-blocking case.

Reply:
did you end up trying this for non-blocking route models?